src/autotester/server/utils/tests/file_management_test.py

```diff
@@ -0,0 +1,221 @@
+from autotester.server.utils.file_management import (
+    clean_dir_name,
+    random_tmpfile_name,
+    recursive_iglob,
+    copy_tree,
+    copy_test_script_files,
+    ignore_missing_dir_error,
+    move_tree,
+    fd_open,
+    fd_lock,
+)
+from autotester.server.utils import redis_management
+import os.path
+import re
+import tempfile
+import shutil
+from autotester.config import config
+import subprocess
+import pytest
+
+FILES_DIRNAME = config["_workspace_contents", "_files_dir"]
+
+@pytest.fixture
+def input1():
+    dir1 = tempfile.mkdtemp()
+    fd, file1 = tempfile.mkstemp(dir=dir1)
+    dir2 = tempfile.mkdtemp()
+    sub_dir = tempfile.mkdtemp(dir=dir2)
+    dir3 = tempfile.mkdtemp()
+    sub_dir1 = tempfile.mkdtemp(dir=dir3)
+    sub_dir2 = tempfile.mkdtemp(dir=sub_dir1)
+    fd, file3 = tempfile.mkstemp(dir=sub_dir2)
+    yield dir1, file1, dir2, sub_dir, dir3, sub_dir1, sub_dir2, file3
+    if os.path.exists(dir1): shutil.rmtree(dir1)
+    if os.path.exists(dir2): shutil.rmtree(dir2)
+    if os.path.exists(dir3): shutil.rmtree(dir3)
+
+
+def list_of_fd(file_or_dir):
+    dir = []
+    files = []
+    for i in file_or_dir:
+        dir.append(i[1]) if i[0] == "d" else files.append(i[1])
+    return dir, files
+
+
+def test_clean_dir_name():
+    a = "markus/address"
+    b = a.replace("/", "_")
+    assert clean_dir_name(a) == b
+
+
+def test_random_tmpfile_name():
+    tmp_file = random_tmpfile_name()
+    dir_name = os.path.dirname(tmp_file)
+    base_name = os.path.basename(tmp_file)
+    assert os.path.exists(dir_name) and re.match(
+        r"[+\-]" "0" "[xX]" "0|" "([1-9A-Fa-f][1-9A-Fa-f]*)", base_name
+    )
+
+
+# When the Directory is empty
+def test_recursive_iglob_1(input1):
```

Comment:
Please don't just enumerate your test functions (add 1, 2, etc. at the end of their name). Either describe what they do in the name of the test function or put similar tests (ie. tests that test the same method) in their own class.

for example:

```python
class TestRecursiveIglob:
    def test_empty_dir(...):
        ...
    def test_has_one_file(...):
        ...
    def test_has_one_subdir(...):
        ...
```